### PR TITLE
Level4

### PR DESCRIPTION
**Pull request by jeaniewu**
Note: Merging in before the changes get way too big (level 4 is not done yet), Also I notice that textboxManager and CameraFollow force player to always have Controller2 on, I made a fix to that in case ur wondering why I change those scripts 

Feature implemented:
- Level 4a (i) and 4a (ii)
- added sliced doll animation (https://trello.com/c/zX2wC4or/46-add-sliced-doll-animation)

Sound implemented:
- https://trello.com/c/FxXjQZs5/15-sawblades
- https://trello.com/c/pKaYk8Xw/11-getting-chopped-in-half-sound

Bugs and Refactoring:
- make sawblade collider skinnier (https://trello.com/c/9iB9QOWA/55-make-sawblade-colider-a-little-skinnier)
- NPE switching doll and then doll dies (https://trello.com/c/oLTt8tBy/56-null-reference-error-when-about-to-switch-to-a-doll-and-the-doll-dies)
- fix ghost switch bug (https://trello.com/c/BP3X0uSv/63-in-level-1a-when-the-two-dolls-are-too-far-away-from-one-another-and-you-try-switching-it-stops-you-from-moving-as-if-trying-to-)

### CODE REVIEW

File: Assets/Scripts/AnimationController.cs [hunk 1]
```diff
@@ -0,0 +1,24 @@
+﻿using UnityEngine;
+using System.Collections;
+
+public class AnimationController : MonoBehaviour {
```
Comment (jayfriso) [inline Assets/Scripts/AnimationController.cs]: Maybe call this DollAnimationController as it seems specific to dolls.

**Comment by jayfriso** [comment]
Looks good to me, just maybe change the name issue. :+1:

**Comment by jeaniewu** [comment]
@jayfriso done, will let @Gowtham100 review it

**Comment by Gowtham100** [comment]
Damn good work Jen :+1: for me...no questions so far

**Pull request merged**